magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL1_to_DL2.py

```diff
@@ -0,0 +1,302 @@
+            run_LST_id = run_str.lstrip("Run")
+            nsb = df_LST[df_LST["LST1_run"] == run_LST_id]["nsb"].tolist()[0]
+            # rounding the NSB to the nearest MC nsb value
+            for j in range(0, len(nsb_list)):
+                if (nsb <= nsb_limit[j + 1]) & (nsb > nsb_limit[j]):
+                    nsb = nsb_list[j]
+
+            # Writing on output .txt file
+            if nsb <= nsb_limit[-1]:
+                with open(
+                    f"{outdir}/{period}_{nsb}_{night_date}.txt",
+                    "a+",
+                ) as file:
+                    file.write(f"{Run}\n")
+
+
+def bash_DL1Stereo_to_DL2(
+    target_dir, source, env_name, cluster, RF_dir, df_LST, MC_v, version, nice
+):
+    """
+    This function generates the bashscript for running the DL1Stereo to DL2 analisys.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source : str
+        Source name
+    env_name : str
+        Conda enviroment name
+    cluster : str
+        Cluster system
+    RF_dir : str
+        Path to the RFs
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+    MC_v : str
+        Version of MC processing
+    version : str
+        Version of the input (stereo subruns) data
+    nice : int or None
+        Job priority
+    """
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+    process_name = source
+    LST_runs_and_dates = f"{source}_LST_runs.txt"
+    LST_date = []
+    for i in np.genfromtxt(LST_runs_and_dates, dtype=str, delimiter=",", ndmin=2):
+        LST_date.append(str(i[0].replace("_", "")))
+    LST_date = list(set(LST_date))
+    Nights_list = np.sort(
+        glob.glob(f"{target_dir}/v{version}/{source}/DL1Stereo/Merged/*")
+    )
+
+    for night in Nights_list:
+        night_date = night.split("/")[-1]
+        outdir = f"{target_dir}/v{__version__}/{source}/DL2/{night_date}/logs"
+        File_list = glob.glob(f"{outdir}/ST*.txt")
+        night_date = night.split("/")[-1]
+        if str(night_date) not in LST_date:
+            continue
+
+        for file in File_list:
+            with open(file, "r") as f:
+                process_size = len(f.readlines()) - 1
+            if process_size < 0:
+                continue
+            nsb = file.split("/")[-1].split("_")[1]
+            period = file.split("/")[-1].split("_")[0]
+            dec = df_LST[df_LST.source == source].iloc[0]["MC_dec"]
+            if np.isnan(dec):
+                print(f"MC_dec is NaN for {source}")
+                continue
+            dec = str(dec).replace(".", "")
+            dec = str(dec).replace("-", "min_")
+            RFdir = f"{RF_dir}/{period}/NSB{nsb}/v{MC_v}/dec_{dec}/"
+            if (not os.path.isdir(RFdir)) or (len(os.listdir(RFdir)) == 0):
```

Comment:
this checks if the directory is there and is not empty, but it could have e.g. only logs that something failed. 
instead you can check
```if len(glob.glob(f"RFdir/*joblib))<3```
in normal situation you should have 12 files (3 types of RFs times 4 combination of telescopes), but for LST1+monoMAGIC you would have only 3)